src/marker.rs

```diff
@@ -1,6 +1,95 @@
+use std::cell::RefCell;
+use std::rc::{Rc, Weak};
+
+use crate::*;
 use serde::{Deserialize, Serialize};
 use wasm_bindgen::prelude::*;
 
+use crate::event;
+
+pub struct MarkerFactory {
+    pub marker: Rc<Marker>,
+    handle: Option<MarkerHandle>,
+}
```

Comment:
I think you followed `MapFactory` for this naming 👍 

But I actually wanted to change XXXFactory to something more intuitive 😅 Do you have any suggestion?

Reply:
Thank you so much! Let me ponder for a while